rust/xaynet-server/src/storage/api.rs

```diff
@@ -0,0 +1,240 @@
+pub trait CoordinatorStorage
+where
+    Self: Clone + Send + Sync + 'static,
+{
+    /// Sets a [`CoordinatorState`].
+    ///
+    /// # Behavior
+    ///
+    /// - If no state has been set yet, set the state and return `Result::Ok(())`.
+    /// - If a state already exists, override the state and return `Result::Ok(())`.
+    async fn set_coordinator_state(&mut self, state: &CoordinatorState) -> StorageResult<()>;
+
+    /// Gets a [`CoordinatorState`].
+    ///
+    /// # Behavior
+    ///
+    /// - If no state has been set yet, return `Result::Ok(Option::None)`.
+    /// - If a state exists, return `Result::Ok(CoordinatorState)`.
+    async fn get_coordinator_state(&mut self) -> StorageResult<Option<CoordinatorState>>;
+
+    /// Adds a sum participant entry to the [`SumDict`].
+    ///
+    /// # Behavior
+    ///
+    /// - If a sum participant has been successfully added, return `Result::Ok(SumPartAdd(Result::Ok(())))`
+    /// - If the participant could not be added due to a PET protocol error, return
+    ///   the corresponding `Result::Ok(SumPartAdd(Result::Err(SumPartAddError)))`.
+    async fn add_sum_participant(
+        &mut self,
+        pk: &SumParticipantPublicKey,
+        ephm_pk: &SumParticipantEphemeralPublicKey,
+    ) -> StorageResult<SumPartAdd>;
+
+    /// Gets the [`SumDict`].
+    ///
+    /// # Behavior
+    ///
+    /// - If the sum dict does not exist, return `Result::Ok(Option::None)`.
+    /// - If the sum dict exists, return `Result::Ok(Option::Some(SumDict))`.
+    async fn get_sum_dict(&mut self) -> StorageResult<Option<SumDict>>;
+
+    /// Adds a local [`LocalSeedDict`] of the given [`UpdateParticipantPublicKey`] to the [`SeedDict`].
+    ///
+    /// # Behavior
+    ///
+    /// - If the local seed dict has been successfully added, return `Result::Ok(LocalSeedDictAdd(Result::Ok(())))`.
+    /// - If the local seed dict could not be added due to a PET protocol error, return
+    ///   the corresponding `Result::Ok(LocalSeedDictAdd(Result::Err(LocalSeedDictAddError)))`.
+    async fn add_local_seed_dict(
+        &mut self,
+        update_pk: &UpdateParticipantPublicKey,
+        local_seed_dict: &LocalSeedDict,
+    ) -> StorageResult<LocalSeedDictAdd>;
+
+    /// Gets the [`SeedDict`].
+    ///
+    /// # Behavior
+    ///
+    /// - If the seed dict does not exist, return `Result::Ok(Option::None)`.
+    /// - If the seed dict exists, return `Result::Ok(Option::Some(SeedDict))`.
+    async fn get_seed_dict(&mut self) -> StorageResult<Option<SeedDict>>;
+
+    /// Increments the mask score with the given [`MaskObject`].
+    /// The score of the given mask is incremented by `1`.
+    ///
+    /// # Behavior
+    ///
+    /// - If the mask score has been successfully incremented, return `Result::Ok(Vec<(MaskObject, u64)>)`.
+    /// - If the mask score could not be incremented due to a PET protocol error, return
+    ///   the corresponding `Result::Ok(MaskScoreIncr(Result::Err(MaskScoreIncrError)))`.
```

Comment:
the described `Vec<_>` return type doesn't fit.

```suggestion
    /// - If the mask score has been successfully incremented, return `StorageResult::Ok(MaskScoreIncr(Result::Ok(())))`.
    /// - If the mask score could not be incremented due to a PET protocol error, return
    ///   the corresponding `StorageResult::Ok(MaskScoreIncr(Result::Err(MaskScoreIncrError)))`.
```

similarly as before, maybe without exposing internals:

```suggestion
    /// - If the mask score has been successfully incremented, return `StorageResult::Ok(MaskScoreIncr(Result::Ok(())))`.
    /// - If the mask score could not be incremented due to a PET protocol error, return
    ///   the corresponding `StorageResult::Ok(MaskScoreIncr(Result::Err(MaskScoreIncrError)))`.
```